orca-keel/src/test/java/com/netflix/spinnaker/orca/keel/task/ImportDeliveryConfigTaskTest.java

```diff
@@ -0,0 +1,442 @@
+            "HTTP 400 "
+                + wireMock.baseUrl()
+                + "/delivery-configs/: Status: 400, URL: "
+                + wireMock.baseUrl()
+                + "/delivery-configs/, Message: Bad Request");
+
+    var errorMap = new HashMap<>();
+    errorMap.put("message", expectedMessage);
+
+    TaskResult terminal =
+        TaskResult.builder(ExecutionStatus.TERMINAL).context(Map.of("error", errorMap)).build();
+
+    // Simulate any 4xx http error with empty error response body
+    String emptyBody = "";
+    simulateFault("/delivery-configs/", emptyBody, HttpStatus.BAD_REQUEST);
+
+    getDeliveryConfigManifest();
+
+    var result = importDeliveryConfigTask.execute(stage);
+
+    verifyGetDeliveryConfigManifestInvocations();
+
+    assertThat(result).isEqualTo(terminal);
+  }
+
+  @Test
+  public void testTaskResultWhenHttp5xxErrorIsThrown() {
+
+    contextMap.put("attempt", (Integer) contextMap.get("attempt") + 1);
+    contextMap.put(
+        "errorFromLastAttempt",
+        "Retryable HTTP response 500 received from downstream service: HTTP 500 "
+            + wireMock.baseUrl()
+            + "/delivery-configs/: Status: 500, URL: "
+            + wireMock.baseUrl()
+            + "/delivery-configs/, Message: Server Error");
+
+    TaskResult running = TaskResult.builder(ExecutionStatus.RUNNING).context(contextMap).build();
+
+    // Simulate any 5xx http error with empty error response body
+    String emptyBody = "";
+    simulateFault("/delivery-configs/", emptyBody, HttpStatus.INTERNAL_SERVER_ERROR);
+
+    getDeliveryConfigManifest();
+
+    var result = importDeliveryConfigTask.execute(stage);
+
+    verifyGetDeliveryConfigManifestInvocations();
+
+    assertThat(result).isEqualTo(running);
+  }
+
+  @Test
+  public void testTaskResultWhenAPIFailsWithNetworkError() {
+
+    contextMap.put("attempt", (Integer) contextMap.get("attempt") + 1);
+    contextMap.put(
+        "errorFromLastAttempt",
+        String.format(
+            "Network error talking to downstream service, attempt 1 of %s: Connection reset: Connection reset",
+            contextMap.get("maxRetries")));
+
+    TaskResult running = TaskResult.builder(ExecutionStatus.RUNNING).context(contextMap).build();
+
+    // Simulate network failure
+    simulateFault("/delivery-configs/", Fault.CONNECTION_RESET_BY_PEER);
+
+    getDeliveryConfigManifest();
+
+    var result = importDeliveryConfigTask.execute(stage);
+
+    verifyGetDeliveryConfigManifestInvocations();
+
+    assertThat(result).isEqualTo(running);
+  }
+
+  /**
+   * Test to verify when the timestamp is null , field will be initialized with default value {@link
+   * Instant#now}
+   */
+  @Test
+  public void testTaskResultWhenTimestampIsNullForSpringHttpError() throws JsonProcessingException {
+
+    var httpStatus = HttpStatus.BAD_REQUEST;
+
+    // SpringHttpError is initialized with no timestamp
+    var httpError = makeSpringHttpError(httpStatus, null);
```

Comment:
It would be nice for this test to appear in the "before" commit as well, so we can see how it behaves.

But also, this way of building a response body is a little indirect.  The point isn't that timestamp is null in SpringHttpError, the point is that the response body doesn't have a timestamp in it.

Reply:
New test added in the first commit : 5d711ab18a2c26c3420c55ffcd89af0c1ed67d4f to verify the case when the error response body doesn't contain timestamp field.